tools/codegen/prim/alias_gen.py

```diff
@@ -0,0 +1,339 @@
+
+# TODO: expand this map as needed
+_base_type_to_sig_cpp_map = {
+    BaseTy.Tensor: 'Tensor&',  # Note: may be const Tensor& (handled below)
+    BaseTy.Dimname: 'Dimname',
+    BaseTy.int: 'int64_t',
+    BaseTy.Scalar: 'const Scalar&'
+}
+
+# Generates the C++ representing a type
+def arg_to_sig_cpp(arg: Argument) -> str:
+    cpp: str
+
+    # Computes the C++ type
+    if isinstance(arg.type, ListType):
+        cpp = list_type_to_sig_cpp(arg)
+    elif isinstance(arg.type, OptionalType):
+        cpp = optional_type_to_sig_cpp(arg)
+    else:
+        assert isinstance(arg.type, BaseType), "Cannot generate signature C++ for a non-Base, non-Optional, non-List type!"
+        cpp = _base_type_to_sig_cpp_map[arg.type.name]
+
+        # Unmodified tensors are passed as const references
+        if arg.type.name is BaseTy.Tensor:
+            is_modified = arg.annotation and arg.annotation.is_write
+            if not is_modified:
+                cpp = "const " + cpp
+
+    # Adds the name
+    cpp += " " + arg.name
+    return cpp
+
+# Returns True when Arguments specifies one or more out arguments, False otherwise
+def has_out(args: Arguments) -> bool:
+    return len(args.out) > 0
+
+# Returns True when Arguments specifies a self arg, False otherwise
+def has_self(args: Arguments) -> bool:
+    return args.self_arg is not None
+
+# Returns True when FunctionSchema specifies an inplace function, False otherwise
+def is_inplace(fs: FunctionSchema) -> bool:
+    return fs.name.name.inplace
+
+# Constructs the C++ for a method call
+def generate_method_call_cpp(fn_cpp: str, args: Arguments) -> str:
+    assert has_self(args), "Request to generate method call C++ with Arguments containing no self arg!"
+    assert not has_out(args), "Request to generate method call C++ with Arguments containing out arg!"
+
+    call_args = list(args.pre_self_positional)
+    call_args.extend(args.post_self_positional)
+    call_args.extend(args.pre_tensor_options_kwarg_only)
+    call_args.extend(args.post_tensor_options_kwarg_only)
+
+    arg_cpp = ", ".join(arg.name for arg in call_args)
+    return "self." + fn_cpp + "(" + arg_cpp + ")"
+
+# Generates the C++ for a function call
+# Note: assumes the function is in the at namespace
+def generate_function_call_cpp(fn_cpp: str, args: Arguments) -> str:
+    call_args = list(args.out)
+    call_args.extend(args.pre_self_positional)
+    if has_self(args):
+        assert args.self_arg is not None  # This is for mypy, which can't figure out the has_self() call
+        call_args.append(args.self_arg.argument)
+    call_args.extend(args.post_self_positional)
+    call_args.extend(args.pre_tensor_options_kwarg_only)
+    call_args.extend(args.post_tensor_options_kwarg_only)
+
+    arg_cpp = ", ".join(arg.name for arg in call_args)
+    return "at::" + fn_cpp + "(" + arg_cpp + ")"
+
+# Generates the C++ for a function signature
+def generate_signature_cpp(return_cpp: str, fn_cpp: str, args: Arguments) -> str:
```

Comment:
`gen.py` already has machinery to generate cpp signature from native function object, see e.g. `compute_registration_declarations` that does that (+ comments that we don't need here), it's better to reuse it 
https://github.com/pytorch/pytorch/blob/71a41323bb42e3b75d259a546a31172d5fe6590c/tools/codegen/gen.py#L914